libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,111 @@
+// SPI HAL Interface
+
+#include "interrupt.h"
+#include "status.h"
+#include "stm32f0xx.h"
+
+typdef void (*spi_callback)(SPIPeriph spi_x, void *context);
+
+// For selecting which peripheral
+typedef enum {
+  SPI_PERIPH_1 = 0,
+  SPI_PERIPH_2,
+} SPIPeriph;
+
+// For setting communication directions
+typedef enum {
+  SPI_DIR_2LINES_FD = 0,
+  SPI_DIR_2LINES_RX_ONLY,
+  SPI_DIR_1LINE_RX,
+  SPI_DIR_1LINE_TX,
+} SPIDir;
+
+// For setting master or slave mode
+typedef enum {
+  SPI_MODE_MASTER = 0,
+  SPI_MODE_SLAVE,
+} SPIMode;
+
+// For setting data size
+typedef enum {
+  SPI_DATASIZE_4B = 0,
+  SPI_DATASIZE_5B,
+  SPI_DATASIZE_6B,
+  SPI_DATASIZE_7B,
+  SPI_DATASIZE_8B,
+  SPI_DATASIZE_9B,
+  SPI_DATASIZE_10B,
+  SPI_DATASIZE_11B,
+  SPI_DATASIZE_12B,
+  SPI_DATASIZE_13B,
+  SPI_DATASIZE_14B,
+  SPI_DATASIZE_15B,
+  SPI_DATASIZE_16B,
+} SPIDataSize;
+
+// For setting clock polarity
+typedef enum {
+  SPI_CPOL_LOW = 0,
+  SPI_CPOL_HIGH,
+} SPICPOL;
```

Comment:
We should figure out how to deal with acronyms and things that involve consecutive capital letters. @ckitagawa @karlding